### PR TITLE
[feature] add vitest

### PR DESCRIPTION
**Pull request by shellyear**
@blcham 

```

 Test Files  39 passed | 5 skipped (44)
      Tests  272 passed | 21 skipped (293)
   Start at  22:12:40
   Duration  10.56s (transform 2.24s, setup 4.24s, collect 18.21s, tests 752ms, environment 11.13s, prepare 2.65s)

```
There are remaining errors that are not shown in the terminal, but in the build pipeline:
[1]
```
_[ This error originated in "tests/__tests__/actions/UserActions.spec.jsx" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯ Uncaught Exception ⎯⎯⎯⎯⎯
AssertionError: expected [ …(3) ] to deeply equal [ { …(2) }, { …(3) }, { …(2) } ]_
```
[2]
```
Error: done() callback is deprecated, use promise instead
 ❯ context node_modules/@vitest/runner/dist/index.js:73:11
 ❯ Timeout._onTimeout tests/__tests__/actions/UserActions.spec.jsx:566:7
    564|     setTimeout(() => {
    565|       expect(store.getActions()).toEqual(expectedActions);
    566|       done();
       |       ^
    567|     }, TEST_TIMEOUT);
    568|   });
 ❯ listOnTimeout node:internal/timers:569:17
 ❯ processTimers node:internal/timers:512:7

This error originated in "tests/__tests__/actions/UserActions.spec.jsx" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.
```
[3] warning
_stderr | tests/__tests__/components/User.spec.jsx > User > renders filled admin's form
Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot_

UPD: all errors are runtime and relate to UserActions.spec.jsx

e.g 
```
stdout | tests/__tests__/actions/UserActions.spec.jsx > User asynchronous actions > creates SAVE_USER_SUCCESS action when saving user successfully is done
{
  storeActions: [ { type: 'SAVE_USER_PENDING', actionFlag: 'CREATE_ENTITY' } ],
  expectedActions: [
    { type: 'SAVE_USER_PENDING', actionFlag: 'CREATE_ENTITY' },
    {
      type: 'SAVE_USER_SUCCESS',
      actionFlag: 'CREATE_ENTITY',
      user: [Object]
    },
    { type: 'LOAD_USERS_PENDING' },
    { type: 'LOAD_USERS_SUCCESS', users: [Array] }
  ]
}
```

### CODE REVIEW

**Comment by shellyear** [comment]
@blcham 
There are runtime errors left, also @cfaester/enzyme-adapter-react-18 [will be removed](https://github.com/kbss-cvut/record-manager-ui/issues/128)

**Comment by blcham** [comment]
xit --> test.skip

**Comment by blcham** [comment]
@shellyear still here there are 57 files committed, you need to rebase from main:
image

It seems that you forgot to do sync before rebase:
image

**Comment by shellyear** [comment]
> @shellyear still there are 57 files committed (i.e. to review), you need to rebase from main: image
> 
> It seems that you forgot to do sync before rebase: image

@blcham  I did it manually using, git fetch upstream & git pull upstream main from the main branch

File: package.json [hunk 1]
```diff
@@ -41,7 +41,6 @@
   "devDependencies": {
     "@babel/eslint-parser": "^7.23.10",
     "@babel/preset-react": "^7.23.3",
-    "@cfaester/enzyme-adapter-react-18": "^0.7.1",
```
Comment (blcham) [inline package.json:44]: TODO close relevant ticket on merge.

File: vite.config.js [hunk 2]
```diff
@@ -48,5 +48,10 @@ export default defineConfig({
     environment: "jsdom",
     globals: true,
     setupFiles: ["./tests/setup.js"],
+    server: {
+      deps: {
+        inline: ["@kbss-cvut/s-forms"],
```
Comment (blcham) [inline vite.config.js:53]: Can we describe why we need this?

**Review by blcham — APPROVED** [review]
LGTM, nicely done!, can be merged after resolved comments

**Review by blcham — CHANGES_REQUESTED** [review]
Note that tests are again failing.

File: vite.config.js [hunk 3]
```diff
@@ -43,4 +44,9 @@ export default defineConfig({
       url: "url-parse", // workaround for parse-link-header library that replaces nodejs builtin module with the module adapted for browser
     },
   },
+  test: {
+    environment: "jsdom",
+    globals: true,
```
Comment (blcham) [inline vite.config.js:49]: Just wondering ... do we still use globals ?

Comment (shellyear) [reply vite.config.js:49]: @blcham  Not anymore, I will remove it

File: package.json (hunk 1 above)
Comment (shellyear) [reply package.json:44]: Done, https://github.com/kbss-cvut/record-manager-ui/issues/128

Comment (blcham) [reply package.json:44]: Ok, I just put explicit reference there.

**Comment by shellyear** [comment]
@blcham fixed the last error https://github.com/kbss-cvut/record-manager-ui/pull/130/commits/03235d8226637d646a71b12f3afabd2bc7d0c920

**Review by blcham — APPROVED** [review]
LGTM, good job!

File: vite.config.js (hunk 2 above)
Comment (blcham) [reply vite.config.js:52]: @shellyear I created https://github.com/kbss-cvut/s-forms/issues/296 but cannot assign you to the ticket yet -- please confirm pending invitation there (should have it in email).

Comment (shellyear) [reply vite.config.js:52]: @blcham accepted the invitation

**Pull request merged**